src/hypertable.c

```diff
@@ -2402,6 +2402,67 @@ ts_hypertable_get_open_dim_max_value(const Hypertable *ht, int dimension_index,
 	return max_value;
 }
 
+/*
+ * Get the min value of an open dimension.
+ */
+int64
+ts_hypertable_get_open_dim_min_value(const Hypertable *ht, int dimension_index, bool *isnull)
+{
+	StringInfo command;
+	const Dimension *dim;
+	int res;
+	bool min_isnull;
+	Datum mindat;
+	Oid timetype;
+
+	dim = hyperspace_get_open_dimension(ht->space, dimension_index);
+
+	if (NULL == dim)
+		elog(ERROR, "invalid open dimension index %d", dimension_index);
+
+	timetype = ts_dimension_get_partition_type(dim);
+
+	/*
+	 * Query for the oldest chunk in the hypertable.
+	 * Anyway to optimize this better?
+	 */
+	command = makeStringInfo();
+	appendStringInfo(command,
+					 "SELECT pg_catalog.min(%s) FROM %s.%s",
+					 quote_identifier(NameStr(dim->fd.column_name)),
+					 quote_identifier(NameStr(ht->fd.schema_name)),
+					 quote_identifier(NameStr(ht->fd.table_name)));
```

Comment:
This does a full table scan of the entire hypertable. With a hypertable that is big, this is going to be a significant problem, *especially* if tiering is involved.

Reply:
Changed this to get min value from chunk metadata.